Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,379 @@
+    (fun _ : ℝ ↦ ‖f z‖) ≤ᶠ[nhdsWithin 0 (Ioi 0)]
+    (fun ε ↦ ‖((ε + sSupNormIm f 0)^(1-z) * (ε + sSupNormIm f 1)^(z) : ℂ)‖ ) := by
+  rw [EventuallyLE, eventually_nhdsWithin_iff, eventually_nhds_iff]
+  use Ioo (-1) 1
+  refine ⟨ ?_, ⟨isOpen_Ioo, by simp only [mem_Ioo, Left.neg_neg_iff, zero_lt_one, and_self]⟩⟩
+  simp only [comp_apply, mem_Ioo, mem_Ioi, and_imp] at *
+  · intro x _ _ hnezero
+    exact norm_le_interp_on_closedStrip_eps f x (hnezero) z hB hd
+      (mem_of_mem_of_subset hz (preimage_mono Ioo_subset_Icc_self))
+
+lemma norm_le_interp_on_strip_zero (hd : DiffContOnCl ℂ f (verticalStrip 0 1))
+    (hB : BddAbove ((norm ∘ f) '' (verticalClosedStrip 0 1))) :
+    ∀ z ∈ verticalStrip 0 1, ‖f z‖ ≤ ‖interpStrip f z‖ := by
+  intro z hz
+  apply tendsto_le_of_eventuallyLE _ _ (eventuallyle f z hB hd hz)
+  · apply tendsto_inf_left
+    simp only [tendsto_const_nhds_iff]
+  -- Proof that we can let epsilon tend to zero.
+  · rw [interpStrip_eq_on_Strip _ _ hz]
+    convert ContinuousWithinAt.tendsto _ using 2
+    · simp only [ofReal_zero, zero_add]
+    · simp_rw [← ofReal_add, norm_eq_abs]
+      have : ∀ x ∈ (Ioi 0), (x + sSupNormIm f 0) ^ (1 - z.re) * (x + (sSupNormIm f 1)) ^ z.re
+          = abs (↑(x + sSupNormIm f 0) ^ (1 - z) * ↑(x + sSupNormIm f 1) ^ (z)) := by
+              intro x hx
+              simp only [map_mul]
+              repeat rw [abs_cpow_eq_rpow_re_of_nonneg (le_of_lt (sSupNormIm_eps_pos f hx _)) _]
+              · simp only [sub_re, one_re]
+              · simpa using (ne_comm.mpr (ne_of_lt hz.1))
+              · simpa [sub_eq_zero] using (ne_comm.mpr (ne_of_lt hz.2))
+      apply tendsto_nhdsWithin_congr this _
+      simp only [zero_add]
+      rw [map_mul, abs_cpow_eq_rpow_re_of_nonneg (sSupNormIm_nonneg _ _) _,
+        abs_cpow_eq_rpow_re_of_nonneg (sSupNormIm_nonneg _ _) _]
+      · apply Tendsto.mul
+        · apply Tendsto.rpow_const
+          · nth_rw 2 [← zero_add (sSupNormIm f 0)]
+            exact Tendsto.add_const (sSupNormIm f 0) (tendsto_nhdsWithin_of_tendsto_nhds
+              (Continuous.tendsto continuous_id' _))
+          · right; simp only [sub_nonneg, le_of_lt hz.2]
+        · apply Tendsto.rpow_const
+          · nth_rw 2 [← zero_add (sSupNormIm f 1)]
+            exact Tendsto.add_const (sSupNormIm f 1) (tendsto_nhdsWithin_of_tendsto_nhds
+              (Continuous.tendsto continuous_id' _))
+          · right; simp only [sub_nonneg, le_of_lt hz.1]
+      · simpa using (ne_comm.mpr (ne_of_lt hz.1))
+      · simpa [sub_eq_zero] using (ne_comm.mpr (ne_of_lt hz.2))
+
+/--
+**Hadamard three-line theorem** on `[0,1]`: If `f` is a bounded function, continuous on `[0,1]`
+and differentiable on `(0,1)`, then for `M(x) := sup ((norm ∘ f) '' (re ⁻¹' {x}))` we have that
+`∀ z ∈ [0,1]` the inequality `|f(z)| ≤ |M(0)^(1-z)| * |M(1)^z|` holds. -/
+lemma norm_le_interp_on_closedStrip (f : ℂ → E) {z : ℂ} (hz : z ∈ verticalClosedStrip 0 1)
```

Comment:
```suggestion
lemma norm_le_interpStrip_of_mem_verticalClosedStrip (f : ℂ → E) {z : ℂ} (hz : z ∈ verticalClosedStrip 0 1)
```
The names should follow the symbols in the statement. Since the statement mentions `interpStrip` and not `interp`, use `interpStrip`. Same thing for `verticalClosedStrip`. 

Note that I think an even more useful version of the statement would be: assume that for all `z` of real part 0, then `||f z|| ≤a`, and for all `z` of real part 1 then `||f z|| ≤ b`. Then `||f z|| ≤ a ^ (1 -z.re) * b ^ z.re`. It's more practical to use because you are sidestepping the definition of the interpolation and of your helper functions.

Reply:
I think this is an excellent idea. I have added it.